101-data-lake-store-encryption-key-vault/azuredeploy.json

```diff
@@ -1,65 +1,121 @@
+  "contentVersion": "0.9.0.0",
+  "parameters": {
+    "dataLakeStoreName": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the Data Lake Store account to create."
+      }
+    },
+    "location": {
+      "type": "string",
+      "allowedValues": [
+        "East US 2",
+        "Central US",
+        "North Europe"
+      ],
+      "defaultValue": "East US 2",
+      "metadata": {
+        "description": "The location in which to create the Data Lake Store account."
+      }
+    },
+    "keyVaultName": {
+      "type": "string",
+      "metadata": {
+        "description": "The Azure Key Vault name."
+      }
+    },
+    "keyName": {
+      "type": "string",
+      "metadata": {
+        "description": "The Azure Key Vault encryption key name."
+      }
+    },
+    "keyVersion": {
+      "type": "string",
+      "metadata": {
+        "description": "The Azure Key Vault encryption key version."
+      }
+    }
+  },
+  "resources": [
+    {
+      "apiVersion": "2016-11-01",
+      "name": "[parameters('dataLakeStoreName')]",
+      "location": "[parameters('location')]",
+      "type": "Microsoft.DataLakeStore/accounts",
+      "properties": {
+        "encryptionState": "Enabled",
+        "encryptionConfig": {
+          "type": "UserManaged",
+          "keyVaultMetaInfo": {
+            "keyVaultResourceId": "[resourceId('Microsoft.KeyVault/vaults', parameters('keyVaultName'))]",
+            "encryptionKeyName": "[parameters('keyName')]",
+            "encryptionKeyVersion": "[parameters('keyVersion')]"
+          }
         }
+      },
+      "identity": {
+        "type": "SystemAssigned"
+      }
+    },
+    {
+      "type": "Microsoft.KeyVault/vaults/accessPolicies",
+      "name": "[concat(parameters('keyVaultName'), '/add')]",
+      "apiVersion": "2015-06-01",
+      "dependsOn": [
+        "[resourceId('Microsoft.DataLakeStore/accounts', parameters('dataLakeStoreName'))]"
+      ],
+      "properties": {
+        "accessPolicies": [
+          {
+            "objectId": "[reference(resourceId('Microsoft.DataLakeStore/accounts', parameters('dataLakeStoreName')), '2016-11-01', 'Full').identity.principalId]",
+            "tenantId": "[reference(resourceId('Microsoft.DataLakeStore/accounts', parameters('dataLakeStoreName')), '2016-11-01', 'Full').identity.tenantId]",
+            "permissions": {
+              "keys": [
+                "encrypt",
+                "decrypt"
+              ]
+            }
+          }
+        ]
+      }
     },
-    "resources": [
-        {
-            "apiVersion": "2016-11-01",
-            "name": "[parameters('accountName')]",
-            "location": "[parameters('location')]",
-            "type": "Microsoft.DataLakeStore/accounts",
-            "properties": {
-                "newTier": "Consumption",
-                "encryptionState": "Enabled",
+    {
+      "type": "Microsoft.Resources/deployments",
```

Comment:
does this need to be nested?

Reply:
Yes; an error is raised when an ARM template modifies the same resource twice.